### PR TITLE
Implements resetUserId()

### PR DESCRIPTION
**Pull request by mattab**
Implements the new resetUserId() method from JS tracker in https://github.com/matomo-org/matomo/pull/12141 https://github.com/matomo-org/matomo/issues/7556

### CODE REVIEW

File: PiwikTracker.php [hunk 1]
```diff
@@ -1163,14 +1163,16 @@ public function setIp($ip)
      *
      * A User ID can be a username, UUID or an email address, or any number or string that uniquely identifies a user or client.
      *
-     * @param string $userId Any user ID string (eg. email address, ID, username). Must be non empty. Set to false to de-assign a user id previously set.
+     * Call resetUserId() to de-assign a user id previously set.
+     *
+     * @param string $userId Any user ID string (eg. email address, ID, username). Must be non empty.
```
Comment (tsteur) [inline PiwikTracker.php]: FYI: The doc may need to be adjusted as when passing `false` it does something possibly unexpected even though it says "Must not be empty".

**Comment by tsteur** [comment]
Since there is https://github.com/matomo-org/matomo-php-tracker/pull/44 this PR should not be needed anymore maybe? Unless you still want to maybe add the `resetUserId ` method (I reckon it is maybe not needed).

**Comment by diosmosis** [comment]
@mattab can you answer @tsteur 's commen?

**Comment by mattab** [comment]
Closing for now, as it's not really needed afaik

**Pull request closed without merging**